### ISSUE TITLE
Xiaomi Mix Fold 2 Issues

### ISSUE THREAD

**Issue by orionbg** [AndyCGYan/lineage_build_unified#53]
Hello and thank you for all of your work!

I have been trying to find a GSI Rom for my Xiaomi Mix Fold 2 for some days.
Pixel Experience seems to be one of the more stable obes but I still do have some issues.
1. Most importantly, I can't receive SMSes! This happens with all GSI Roms I have tried. I can send SMSes without problems but can't receive.
2.  VoWifi not working. While using the alternative IMS.apk did activate VoLTE for me and that works properly, I can't seem to make VoWifi to work. My operator does support it and when I put my SIM card in another device, I have VoWifi working.
3. Since my phone is a foldable, I have two screens. The brightness control on the external screen does not work at all. Also, there is no auto-brightness control on the internal screen (manual brightness control works).
4. The external screen some times will not wake up especially if AOD is active.
5. accessing Doze Features in the PHH Settings menu, crashes the settings app 99% of the times. In the other 1% it crashes when I toggle one of the settings in the menu.
6. Again in the Phh Settings menu, in the section Xiaomi Features, the option DT2W works only on the internal screen.
So basically, two quite important things (SMS and VoWifi) and the rest are secondary.
I'd be happy to test any fixes you can think of.
Also, there is no TWRP for this device so The only method for loading things is Magisk (or manually )...
Thanks!

**Comment by AndyCGYan** [comment]
Someone that I know of has brought up proper device-specific LOS on Mix Fold 2, and it might just become available at some point.

SMS reception can probably be worked around; I believe VoWiFi was removed from our IMS ports.

**Comment by AndyCGYan** [comment]
https://t.me/hawaiireleases - keyword "#zizhan"
